test/box-tap/gh-6647-yield-gc-hook.test.lua

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env tarantool
+
+local tap = require('tap')
+local test = tap.test('gh-6647-yield-gc-hook')
+local fio = require('fio')
+test:plan(1)
+
+local tarantool_bin = arg[-1]
+local PANIC = 256
+
+local function run_script(code)
+    local dir = fio.tempdir()
+    local script_path = fio.pathjoin(dir, 'script.lua')
+    local script = fio.open(script_path, {'O_CREAT', 'O_WRONLY', 'O_APPEND'},
+        tonumber('0777', 8))
+    script:write(code)
+    script:write("\nos.exit(0)")
+    script:close()
+    local cmd = [[/bin/sh -c 'cd "%s" && "%s" ./script.lua 2> /dev/null']]
```

Comment:
How about writing stderr to a file and check that it contains `'fiber [0-9]\+ is switched while running GC finalizer'`? Or, maybe better, set `box.cfg({log = 'script.log'})` and find the pattern in the file.